src/vm/jitinterface.cpp

```diff
@@ -1302,9 +1302,23 @@ struct TryResolveTokenFilterParam
 
 bool isValidTokenForTryResolveToken(CEEInfo* info, CORINFO_RESOLVED_TOKEN* resolvedToken)
 {
-    LIMITED_METHOD_CONTRACT;
+    CONTRACTL {
+        NOTHROW;
+        GC_TRIGGERS;
```

Comment:
It may also be the case that the contracts on `isValidToken` are overly conservative: `IsDynamicScope` is `LIMITED_METHOD_CONTRACT`, `Module::GetMDImport` wraps `PEFile::GetPersistentMDImport` which is also `LIMITED_METHOD_CONTRACT`, and `isValidToken` appears to lack any contracts (assuming it's backed by the implementation in `mdinternalro.cpp`). Is the latter forcing the currently applied contracts?